spec/features/hiring_staff_can_edit_a_draft_vacancy_spec.rb

```diff
@@ -7,223 +7,285 @@
+          # We use the session to store vacancy attributes, make sure it doesn't leak between edits.
+          before do
+            visit edit_organisation_job_path(id: draft_vacancy.id)
 
-        draft_vacancy.salary = 'Pay scale 1 to Pay scale 2'
-        draft_vacancy.benefits = 'Gym, health insurance'
+            draft_vacancy.salary = 'Pay scale 1 to Pay scale 2'
+            draft_vacancy.benefits = 'Gym, health insurance'
 
-        fill_in_pay_package_form_fields(draft_vacancy)
-        click_on I18n.t('buttons.continue')
+            fill_in_pay_package_form_fields(draft_vacancy)
+            click_on I18n.t('buttons.continue')
 
-        draft_vacancy.starts_on = DateTime.now + 1.year
-        draft_vacancy.expires_on = draft_vacancy.starts_on - 1.day
-        draft_vacancy.expiry_time = Time.zone.now
-        draft_vacancy.publish_on = DateTime.now + 1.day
+            draft_vacancy.starts_on = DateTime.now + 1.year
+            draft_vacancy.expires_on = draft_vacancy.starts_on - 1.day
+            draft_vacancy.expiry_time = Time.zone.now
+            draft_vacancy.publish_on = DateTime.now + 1.day
 
-        fill_in_important_dates_fields(draft_vacancy)
-        click_on I18n.t('buttons.continue')
+            fill_in_important_dates_fields(draft_vacancy)
+            click_on I18n.t('buttons.continue')
 
-        select_no_for_supporting_documents
-        click_on I18n.t('buttons.continue')
+            select_no_for_supporting_documents
+            click_on I18n.t('buttons.continue')
 
-        edit_a_published_vacancy
-      end
+            edit_a_published_vacancy
+          end
+
+          scenario 'then editing the draft redirects to incomplete step' do
+            visit organisation_job_path(id: draft_vacancy.id)
+            expect(page).to have_content(I18n.t('jobs.current_step', step: 5, total: 7))
+          end
 
-      scenario 'then editing the draft redirects to incomplete step' do
-        visit organisation_job_path(id: draft_vacancy.id)
-        expect(page).to have_content(I18n.t('jobs.current_step', step: 5, total: 7))
+          def edit_a_published_vacancy
+            published_vacancy = create(:vacancy, :published, school: school)
+            visit edit_organisation_job_path(published_vacancy.id)
+            click_header_link(I18n.t('jobs.application_details'))
+
+            fill_in 'application_details_form[application_link]', with: 'https://example.com'
+            click_on I18n.t('buttons.update_job')
+
+            expect(page.body).to include(
+              I18n.t('messages.jobs.listing_updated', job_title: published_vacancy.job_title)
+            )
+          end
+        end
       end
 
-      def edit_a_published_vacancy
-        published_vacancy = create(:vacancy, :published, school: school)
-        visit edit_organisation_job_path(published_vacancy.id)
-        click_header_link(I18n.t('jobs.application_details'))
+      context 'as a SchoolGroup-level user' do
```

Comment:
GitHub hid this file for being a large diff. That was mainly from indentation changes.

Here I added meaningful diff.